pkg/scheduler/core/generic_scheduler.go

```diff
@@ -1094,25 +1096,38 @@ func selectVictimsOnNode(
 	potentialVictims := util.SortableList{CompFunc: util.MoreImportantPod}
 	nodeInfoCopy := nodeInfo.Clone()
 
-	removePod := func(rp *v1.Pod) {
-		nodeInfoCopy.RemovePod(rp)
+	removePod := func(rp *v1.Pod) error {
+		if err := nodeInfoCopy.RemovePod(rp); err != nil {
+			klog.Warning(err)
+			return err
+		}
 		if meta != nil {
-			meta.RemovePod(rp, nodeInfoCopy.Node())
+			if err := meta.RemovePod(rp, nodeInfoCopy.Node()); err != nil {
```

Comment:
Ditto. It's a transaction opertion, if we catch error here, we should rollback the previous nodeInfoCopy.RemovePod(rp).